internal/utils/const.go

```diff
@@ -0,0 +1,18 @@
+package utils
+
+import (
+	"errors"
+	"regexp"
+	"time"
+)
+
+// ErrTimeoutReached error thrown when a timeout is reached
+var ErrTimeoutReached = errors.New("timeout reached")
+var parseDurationRegexp = regexp.MustCompile(`^(?P<Duration>[1-9]\d*?)(?P<Unit>[smhdwMy])$`)
```

Comment:
you can actually do the parsing with only that regexp, no need for `parseDurationSecondsRegexp`. Just put the unit as an optional part with `?` in that regexp and check the number of matches like you already do to know whether the match is in seconds or in custom unit.